src/components/footer/join-newsletter/join-newsletter.tsx

```diff
@@ -3,6 +3,29 @@ import { SingleInputField } from "../single-input-field/single-input-field";
 import styles from "./join-newsletter.module.scss";
 
 export const JoinNewsletter = () => {
+  const [emailSent, setEmailSent] = useState(false);
+
+  const onHandleSubmit = async (value: string) => {
+    try {
+      const response = await fetch("http://localhost:3000/marcin", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ email: value }),
+      });
+
+      if (response.ok) {
+        setEmailSent(true);
+        console.log("Email sent successfully!");
+      } else {
+        console.error("Failed to send email");
+      }
+    } catch (error) {
+      console.error("Error sending email:", error);
+    }
+  };
```

Comment:
Remove console.logs, they should not be present in Pull Request, 
Secondly move this whole functionality to: `api/newsletter.ts` const postNewsletter = ()=>{}
Third, catching the error means something went wrong so we should notify the user, if we are setting setEmailSent we should also show Error to the user `Coś poszło nie tak` or anything like this

Reply:
Secondly: i have setter in this function :( I'll get everything mixed up
Third: i have console.error here. Its not ok?